src/corelibs/U2Core/src/util/FileAndDirectoryUtils.cpp

```diff
@@ -142,4 +143,13 @@ QString FileAndDirectoryUtils::getAbsolutePath(const QString &filePath) {
     return QFileInfo(result).absoluteFilePath();
 }
 
+bool FileAndDirectoryUtils::isDirectoryWritable(const QString &path) {
```

Comment:
We already have a method like this:

bool TmpDirChecker::checkWritePermissions(const QString &dirPath) 
We can move that method's code here.

Reply:
Moved this method from TmpDirChecker to FileAndDirectoryUtils because there it more easy to find.